"Releasing Mods" wiki page + some general tweaks and fixes (#22)
KristalTeam/Shadow: app/wiki/releasing-mods/page.mdx

<summary>
Windows:
</summary>

Windows has quick compression support built into File Explorer. 

Open your mod, and **select everything in the root folder**, right click, and then select:
- `Compress To... > ZIP Archive` (Windows 11)
- `Send To > Compressed (zipped) Folder` (Windows 10)

</details>

<details>

<summary>
Mac:
</summary>

Mac has a built-in zip compression option.

Open your mod, **select everything**, and then right click or ctrl+click and select `Compress n items`.

Alternatively, open a terminal and navigate into your mod's folder, and run:

```cmd
zip -9 -r <YourModName>.zip .
```

</details>

<details>

<summary>
Linux:
</summary>

Open the command line and navigate into your mod's folder. You can then use `zip` to create an archive:

```cmd
zip -9 -r <YourModName>.zip .
```

</details>

That's it! The ZIP file you created contains your mod, ready for sharing!

---
### Steps for Players

If you release your mod on it's own, you should give your players some additional information:

- Link them to the [download guide](/wiki/downloading) and [playing guide](/wiki/playing-mods) for Kristal.
- List the Kristal version you made the mod on.

---
### Compatibility

> Mods released on their own are compatible with all OSes that Kristal supports out of the box, which are in turn the desktop OSes supported by love2d 11.5:

Reviewer: Nitpick: s/love2d/LÖVE for cases like this

---
### Making a .love Archive [love-package]

A `.love` archive is actually just a `.zip` archive (shocked face). That's great for us, because it means it's very simple to make.

#### Step 1

Compress your source code into a **ZIP Archive**:

<details>

<summary>
Windows:
</summary>

Windows has quick compression support built into File Explorer. 

Open your source code and **select everything**, right click, and then select:
- `Compress To... > ZIP Archive` (Windows 11)
- `Send To > Compressed (zipped) Folder` (Windows 10)

</details>

<details>

<summary>
Mac:
</summary>

Mac has a built-in zip compression option.

Open your source code, **select everything**, and then right click or ctrl+click and select `Compress n items`.

Alternatively, open a terminal and navigate into the source code folder, and run:

```cmd
zip -9 -r <YourModName>.zip .
```

</details>

<details>

<summary>
Linux:
</summary>

Open the command line and navigate inside the source code folder. You can then use `zip` to create an archive:

```cmd
zip -9 -r <YourModName>.zip .
```

</details>

When you've got your compressed archive, **change the extension from `.zip` to `.love`**.

---
#### Step 2

> Run your new `.love` archive with love 11.5 to check that it works.

Reviewer: ```suggestion
Run your new `.love` archive with LÖVE 11.5 to check that it works.
```

> If you get an immediate error `"no code to run"`, then you have likely compressed the outer folder of the source code as well. Repeat step 1, making sure to **only copy the contents of the source, and not the containing folder**. 

Reviewer: ```suggestion
If you get an immediate error saying `"no code to run"`, then you have likely compressed the outer folder of the source code as well. Repeat step 1, making sure to **only copy the contents of the source, and not the containing folder**. 
```

Otherwise, if everything has loaded correctly, your archive is ready!

---
#### Compatibility

As previously mentioned, a `.love` archive is compatible with all OSes that are supported by love2d 11.5:

- Windows (Vista and above)
- MacOS (10.11 El Capitan and above)
- Linux

---
#### Steps for Players

> Players will need to download a copy version of love2d 11.5 for their OS to run your mod from a `.love`:

Reviewer: ```suggestion
Players will need to install LÖVE 11.5 to their system in order to run your mod from the `.love` archive:
```

> - Link them to [LOVE's main page](https://www.love2d.org/), and tell them to download and install version `11.5` for their OS.

Reviewer: (nit)
```suggestion
- Link them to [LÖVE's main page](https://love2d.org/), and tell them to download and install version 11.5 for their OS.
```

- Once love is installed, they should be able to open the `.love` archive directly, or by running `love <your-mod-name>.love` in the command line.

---
### Making Fused Executables [fused-executables]

After making a `.love` archive, you can consider making fused executables for some platforms. These combine the `.love` and LOVE itself together, so the user only has to download your mod and run it like an application.

When you distribute your game as a fused executable, **you must include the love2d license file, `license.txt` in your distribution.**

Fused executables are platform specific.

<details>

<summary>
Windows Executables:
</summary>

> Building a windows executable will require the `love2d-11.5` package for windows.

Reviewer: ```suggestion
Building a windows executable will require the `love2d-11.5` package for Windows.
```

---

Your MacOS fused release is ready! To share this build, you'll want to distribute the archive you just created.

You should also check that it works if you're on MacOS, by trying to extract and run it yourself.

</details>

---

<details>

<summary>
Linux AppImages:
</summary>

AppImage is a format for packaging applications on Linux. LOVE has an official AppImage for Linux, which is capable of being fused with `.love` files.

This is a more involved process than regular fusing, so it's best done by following the [official LOVE Distribution For Linux guide](https://love2d.org/wiki/Game_Distribution#Distribution_for_Linux). You should follow the "Official AppImage" subsection, with your "game" being the `.love` archive you created earlier.

</details>
---
#### Compatibility

Fused executables are only compatible with the platform they have been built for.

#### Steps for Players

Playing mods from a fused executable is very easy:
- Extract the downloaded archive.
- Run the executable/application.

</Box>

<Box>
## Distributing

With all your preferred releases for your mod ready, all that's left to do is release them!

> You can share your work anywhere, but if you're looking for Kristal-specific spaces, there is a [Kristal Gamebanana](https://gamebanana.com/games/16251), as well as the [Kristal Discord Server](https://discord.gg/8ZGuKXJE2C) which has it's own `mod-releases` forum.

Reviewer: ```suggestion
You can share your work anywhere, but if you're looking for Kristal-specific spaces, there is a [Kristal GameBanana](https://gamebanana.com/games/16251), as well as the [Kristal Discord Server](https://discord.gg/8ZGuKXJE2C) which has it's own `mod-releases` forum.
```
